sw/host/ot_certs/src/cbor.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, Debug, Deserialize, PartialEq, Eq, Serialize)]
+#[repr(u8)]
+pub enum MajorType {
+    Uint = 0,
+    Sint = 1,
+    Bstr = 2,
+    Tstr = 3,
+    Array = 4,
+    Map = 5,
+}
+
+pub fn arg_size(arg: u64) -> u64 {
+    if arg <= 23 {
+        0
+    } else if arg <= u8::MAX.into() {
+        1
+    } else if arg <= u16::MAX.into() {
+        2
+    } else if arg <= u32::MAX.into() {
+        4
+    } else {
+        8
+    }
+}
+
+fn header(kind: MajorType, arg: u64) -> Vec<u8> {
+    let kind = kind as u8;
+    let first_byte = |value| {
+        assert!(value <= 31);
+        kind << 5 | (value as u8)
+    };
+
+    let size = arg_size(arg);
+    let b0 = ((arg >> 56) & 0xff) as u8;
+    let b1 = ((arg >> 48) & 0xff) as u8;
+    let b2 = ((arg >> 40) & 0xff) as u8;
+    let b3 = ((arg >> 32) & 0xff) as u8;
+    let b4 = ((arg >> 24) & 0xff) as u8;
+    let b5 = ((arg >> 16) & 0xff) as u8;
+    let b6 = ((arg >> 8) & 0xff) as u8;
+    let b7 = (arg & 0xff) as u8;
+
+    if size == 0 {
+        vec![first_byte(arg)]
+    } else if size == 1 {
+        vec![first_byte(24), b7]
```

Comment:
what is this magic value 24? Constants should be defined as such using `const MAGIC_VALUE = 24;` with a reference to the spec if possible.

Reply:
I make them represented as a enum and put some document there.